configuration/configuration_heavy.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configuration
+
+import (
+	"github.com/insolar/insconfig"
+)
+
+// ConfigHeavyBadger contains configuration params for HeavyBadger
+type ConfigHeavyBadger struct {
+	GenericConfiguration `mapstructure:",squash" yaml:",inline"`
+	Ledger               Ledger
+	Exporter             Exporter
+}
+
+// ConfigHeavyPg contains configuration params for HeavyPg
+type ConfigHeavyPg struct {
+	GenericConfiguration `mapstructure:",squash" yaml:",inline"`
+	Ledger               LedgerPg
+	Exporter             Exporter
+}
+
+func (c ConfigHeavyBadger) GetConfig() interface{} {
+	return &c
+}
+
+func (c ConfigHeavyPg) GetConfig() interface{} {
+	return &c
+}
+
+// NewConfigurationHeavyBadger creates new default configuration
+func NewConfigurationHeavyBadger() ConfigHeavyBadger {
+	return ConfigHeavyBadger{
+		Ledger:               NewLedger(),
+		Exporter:             NewExporter(),
+		GenericConfiguration: NewGenericConfiguration(),
+	}
+}
+
+// NewConfigurationHeavyBadger creates new default configuration
+func NewConfigurationHeavyPg() ConfigHeavyPg {
+	cfg := ConfigHeavyPg{
+		Ledger:               NewLedgerPg(),
+		Exporter:             NewExporter(),
+		GenericConfiguration: NewGenericConfiguration(),
+	}
+	return cfg
+}
+
+// HolderHeavyBadger provides methods to manage heavy configuration
+type HolderHeavyBadger struct {
+	Configuration ConfigHeavyBadger
+	Params        insconfig.Params
+}
+
+func (h HolderHeavyBadger) GetGenericConfig() GenericConfiguration {
+	return h.Configuration.GenericConfiguration
+}
+func (h HolderHeavyBadger) GetNodeConfig() interface{} {
+	return h.Configuration
+}
+
+// MustLoad wrapper around Load function which panics on error.
+func (h *HolderHeavyBadger) MustLoad() *HolderHeavyBadger {
+	err := h.Load()
+	if err != nil {
+		panic(err)
```

Comment:
is it OK to panic here?

Reply:
ok, but we need more info, like "can't load config file %s"

Reply:
ok, it's only for tests